datahub-web-react/src/app/ingestV2/source/multiStepBuilder/IngestionSourceUpdatePage.tsx

```diff
@@ -49,40 +65,92 @@ export function IngestionSourceUpdatePage() {
 
     const onSubmit = useCallback(
         async (data: SourceBuilderState | undefined) => {
-            if (data) {
+            if (!data) return undefined;
+
+            const shouldRun = true; // TODO:: set a real value
+            try {
                 const source = ingestionSourceData?.ingestionSource as IngestionSource | undefined;
-                await updateIngestionSource(
-                    urn,
-                    {
-                        type: data.type as string,
-                        name: data.name as string,
+                const input = {
```

Comment:
is it possible to combine some of this logic between `IngestionSourceCreatePage` and `IngestionSourceUpdatePage`?